src/components/TabPaneTabContextMenu.tsx

```diff
@@ -0,0 +1,85 @@
+import { Item, ItemParams, Menu, PredicateParams, Separator } from 'react-contexify';
+import { VscClose, VscCloseAll, VscLayoutSidebarLeft, VscLayoutSidebarRight } from 'react-icons/vsc';
+
+import { useOpenEditorsDataForPane } from '../atoms/hooks/useOpenEditorsDataForPane';
+import { EditorId } from '../atoms/types/EditorData';
+import { PaneId } from '../atoms/types/PaneGroup';
+import { emitEvent } from '../events';
+
+export const TABPANE_TAB_MENU_ID = 'TabPaneContextMenu';
+
+export interface TabPaneTabContextMenuProps {
+  editorId: EditorId;
+  paneId: PaneId;
+}
+
+export function TabPaneTabContextMenu() {
+  const leftOpenEditorsData = useOpenEditorsDataForPane('LEFT');
```

Comment:
You can greatly simplify the logic here since the props contain the `paneId`: there is no need to fetch `leftOpenEditorsData` to check whether the editor is in the left or right editor, you can just use the props attribute